source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,166 @@ namespace aspect
+
+    FEValues<dim> fe_values_21 (mapping,
+                                finite_element,
+                                quadrature_formula_21,
+                                update_values   |
+                                update_quadrature_points |
+                                update_JxW_values);
+    std::vector<double> values_21 (n_q_points_21);
+
+    const FEValuesExtractors::Scalar field
+      = (advection_field.is_temperature()
+         ?
+         introspection.extractors.temperature
+         :
+         introspection.extractors.compositional_fields[advection_field.compositional_variable]
+        );
+
+
+    const double max_advection_exact_global = (advection_field.is_temperature()
+                                               ?
+                                               parameters.global_temperature_max_preset
+                                               :
+                                               parameters.global_composition_max_preset[advection_field.compositional_variable]
+                                              );
+    const double min_advection_exact_global = (advection_field.is_temperature()
+                                               ?
+                                               parameters.global_temperature_min_preset
+                                               :
+                                               parameters.global_composition_min_preset[advection_field.compositional_variable]
+                                              );
+
+
+    LinearAlgebra::BlockVector distributed_vector (introspection.index_sets.system_partitioning,
+                                                   mpi_communicator);
+    const unsigned int block_idx = advection_field.block_index(introspection);
+    distributed_vector.block(block_idx) = solution.block(block_idx);
+
+    std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
+
+    typename DoFHandler<dim>::active_cell_iterator
+    cell = dof_handler.begin_active(),
+    endc = dof_handler.end();
+    for (; cell != endc; ++cell)
+      {
+        if (cell->is_locally_owned())
+          {
+            cell->get_dof_indices (local_dof_indices);
+            fe_values_12.reinit (cell);
+            fe_values_21.reinit (cell);
+
+            fe_values_12[field].get_function_values(solution, values_12);
+            fe_values_21[field].get_function_values(solution, values_21);
+
+            //Find the local max and local min
+            double min_advection_local =  std::numeric_limits<double>::max();
+            double max_advection_local = -std::numeric_limits<double>::max();
+            for (unsigned int q = 0; q < n_q_points_12; ++q)
+              {
+                min_advection_local = std::min<double> (min_advection_local, values_12[q]);
+                max_advection_local = std::max<double> (max_advection_local, values_12[q]);
+              }
```

Comment:
This all can actually be written in a more elegant and shorter way as follows:
```
  const double min_advection_local = std::min (*std::min_element (values_12.begin(), values_12.end()),
                                                                           *std::min_element (values_21.begin(), values_21.end());
```
and similar for the max of course.